reactioncommerce/reaction
# refactor: Remove legacy social plugin, PDP, product grid and WYSIWYG code

Resolves #5393  
Impact: **major**  
Type: **performance|refactor**

## Issue
First of all, the Operator UI still has a Social section, which used to serve the purpose of adding social "share" buttons on the PDP, back in the 1.x days. This is not used anymore and this feature being present in Catalyst may confuse newcomers.

Second, lots of unused code relative to the legacy PDP, the product grid and the old WYSIWYG admin is still in this repo.

## Solution
We need to remove the unused parts, which will help reduce the client bundle size, improving both Catalyst and customer login UI loading times by a little bit. Not to mention a small gain in DX as well.

## Breaking changes
- The `social` plugin is no more.
- The following legacy components were removed:
  - `productDetail`
  - `ProductField`
  - `ProductTags`
  - `ProductMetadata`
  - `PriceRange`
  - `VariantList`
  - `SocialContainer`
  - `VariantListContainer`
  - `ProductNotFound`
- In the `default-theme` plugin, all the CSS files under the `products` directory were removed, with the exception of `products/variantForm.less` which still has literally 3 rules that are still being used.

## Testing
1. Open the Operator UI
2. Notice that everything works just as it used to
3. Notice that the client bundle is a bit lighter

Before:
<img width="603" alt="Screen Shot 2019-07-31 at 15 32 26" src="https://user-images.githubusercontent.com/45489203/62208711-b1c70a80-b3a8-11e9-8b75-dfd6500d0191.png">

After:
<img width="772" alt="Screen Shot 2019-07-31 at 14 48 52" src="https://user-images.githubusercontent.com/45489203/62208726-bab7dc00-b3a8-11e9-9eef-cb707437fac6.png">
